cmd/kraft/setup/setup.go

```diff
@@ -0,0 +1,148 @@
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+package setup
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/shirou/gopsutil/v3/host"
+	"github.com/spf13/cobra"
+
+	"kraftkit.sh/cmdfactory"
+)
+
+type Setup struct{}
+
+func New() *cobra.Command {
+	return cmdfactory.New(&Setup{}, cobra.Command{
+		Short:   "Setup the working environment for building and running unikernels",
+		Use:     "setup",
+		Aliases: []string{"s"},
+		Args:    cmdfactory.MaxDirArgs(1),
+		Long: heredoc.Doc(`
+		Setup the working environment for building and running unikernels`),
+		Example: heredoc.Doc(`
+			# Setup the environment
+			$ kraft setup`),
+		Annotations: map[string]string{
+			"help:group": "build",
+		},
+	})
+}
+
+func (opts *Setup) Run(cmd *cobra.Command, args []string) error {
+	var err error
+
+	os_struct, err := host.Info()
+	if err != nil {
+		return err
+	}
+
+	virt, role, err := host.Virtualization()
+	if err != nil {
+		return err
+	}
+
+	check_os(os_struct.Platform, virt, role)
+	check_virt(os_struct.Platform, virt, role)
+
+	return err
+}
+
+func check_os(os_name string, virt string, role string) error {
+	req := []string{
+		"build-essential",
+		"libncurses-dev",
+		"libyaml-dev",
+		"flex",
+		"git",
+		"wget",
+		"socat",
+		"bison",
+		"unzip",
+		"uuid-runtime",
+		"python3",
+		"python3-setuptools",
+		"python3-pip",
+		"qemu-kvm",
+		"qemu-system-x86",
+		"sgabios",
+	}
+
+	is_root := false
+	var cmd string
+	var args []string
+
+	// TODO add requirements installation for other OSs
+	if strings.Compare(os_name, "debian") == 0 || strings.Compare(os_name, "ubuntu") == 0 {
+		if os.Geteuid() == 0 {
+			is_root = true
+		}
+
+		if is_root != true {
+			cmd = "sudo"
+			args = []string{"apt", "install", "-y"}
+			args = append(args, req...)
+		} else {
+			cmd = "apt"
+			args = []string{"install", "-y"}
+			args = append(args, req...)
+		}
+	}
+
+	var err error
+
+	installCmd := exec.Command(cmd, args...)
+	installCmd.Stdout = os.Stdout
+	installCmd.Stderr = os.Stderr
+	err = installCmd.Run()
+
+	return err
+}
+
+func check_virt(os_name string, virt string, role string) error {
```

Comment:
The default syntax for golang is camelcase, is this intentional? 😄

Reply:
nope :)